modules/fedimint-ln-client/src/lib.rs

```diff
@@ -966,53 +979,75 @@ impl LightningClientModule {
         Ok(federation_meta)
     }
 
-    /// Gateways actively registered with the fed
+    /// Fetches gateways registered with the fed and applies gateway vetting
+    /// from 'vetted_gateways` meta field if configured
     pub async fn fetch_registered_gateways(
         &self,
     ) -> anyhow::Result<Vec<LightningGatewayAnnouncement>> {
         let mut gateways = self.module_api.fetch_gateways().await?;
 
         if !gateways.is_empty() {
-            debug!("Fetching meta overrides from remote source/cache");
             let config = self.client_ctx.get_config().clone();
-            let federation_meta = match self.fetch_meta_overrides(config).await {
-                Ok(meta) => {
-                    if meta.is_empty() {
-                        debug!("No meta overrides found");
-                        return Ok(gateways);
+            let global_meta: BTreeMap<String, String> = config.global.meta.clone();
+            let federation_id = config.global.federation_id();
+
+            let meta = match FederationInfo::from_config(config)
+                .await?
+                .meta::<String>(META_OVERRIDE_URL_KEY)?
```

Comment:
This looks like it might do more work than necessary, I assume you use this construct for the `meta` helper decoding fn? We could instead add a similar helper to the global config struct, which you already access a few lines above.